src/main/java/teammates/ui/controller/InstructorStudentRecordsPageData.java

```diff
@@ -1,24 +1,92 @@
+import teammates.common.datatransfer.StudentAttributes;
 import teammates.common.datatransfer.StudentProfileAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.ui.template.Comment;
+import teammates.ui.template.CommentsForStudentsTable;
+import teammates.ui.template.StudentProfile;
 
 public class InstructorStudentRecordsPageData extends PageData {
 
-    public String courseId;
-    public InstructorAttributes currentInstructor;
-    public StudentProfileAttributes studentProfile;
-    public List<CommentAttributes> comments;
-    public List<SessionAttributes> sessions;
-    public String showCommentBox;
+    private String courseId;
+    private String studentName;
+    private String studentEmail;
+    private String showCommentBox;
+    private StudentProfile studentProfile;
+    public StudentProfileAttributes spa; // used for testing admin message
+    private List<CommentsForStudentsTable> commentsForStudentTable;
+    private List<String> sessionNames;
 
-    public InstructorStudentRecordsPageData(AccountAttributes account) {
-        super(account);
+    public InstructorStudentRecordsPageData(AccountAttributes account, StudentAttributes student,
+                                            String courseId, String showCommentBox, StudentProfileAttributes spa,
+                                            List<CommentAttributes> comments, List<String> sessionNames,
+                                            InstructorAttributes instructor) {
+        super(account, student);
+        this.courseId = courseId;
+        this.studentName = Sanitizer.sanitizeForHtml(student.name);
+        this.studentEmail = student.email;
+        this.showCommentBox = showCommentBox;
+        if (spa != null) {
+            this.spa = spa;
+            String pictureUrl = getPictureUrl(spa.pictureKey);
+            this.studentProfile = new StudentProfile(student.name, spa, pictureUrl);
+        }
+        List<Comment> commentDivs = new ArrayList<Comment>();
+        for (CommentAttributes comment : comments) {
+            Comment commentDiv = new Comment(comment, "You", student.name + " (" + student.team + ", " + student.email + ")");
+            String whoCanSeeComment = getTypeOfPeopleCanViewComment(comment);
+            commentDiv.setVisibilityIcon(whoCanSeeComment);
+            commentDiv.setEditDeleteEnabled(false);
+            commentDiv.setNotFromCommentsPage(student.email);
+            commentDiv.setNumComments(comments.size());
+            commentDivs.add(commentDiv);
+        }
+        this.commentsForStudentTable = new ArrayList<CommentsForStudentsTable>();
+        CommentsForStudentsTable commentsForStudent = new CommentsForStudentsTable("You", commentDivs);
+        commentsForStudent.setInstructorAllowedToGiveComment(instructor.isAllowedForPrivilege(student.section,
+                                        Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS));
```

Comment:
Maybe break at `instructor.isAllowedForPrivilege`? It's not very clear that the constant belongs to the `isAllowedForPrivilege` method call.